### PR TITLE
Move away from photo-centric external resource support into file,image,audio,video

### PR DESCRIPTION
**Pull request by nirvn**
See https://github.com/opengisch/QField/pull/3851 on the QField side and the recent work done in QGIS itself such as https://github.com/qgis/QGIS/pull/51594.

### CODE REVIEW

File: layer.py
Comment (nirvn) [inline layer.py:91]: See https://github.com/opengisch/QField/pull/3851/files#diff-ad0acdd865d66f2f32f6ac8d65b8ce09d91d5b45e79416e408bd8809896c5a2fR166 on how the # ~ tags ~ # are used.

**Comment by suricactus** [comment]
Is there a reason you preferred "resouce" over attachment. At least in the code I wrote so far I tend to stick to "attachment".

**Comment by nirvn** [comment]
@suricactus , QgsExternalResources* 

I agree Attachment is a friendlier keyword, and I'm using it in the frontend of things (e.g.: https://github.com/opengisch/qfieldsync/pull/479/files#diff-cb0fe03aac5d043fa6713776d09aaa241751d0e5b4f9d8d7c2ee1b20d58a6905R85)

But in QGIS, we have some frozen API class names, and I would like to stick to resources here (as well as work done in QField) for that.

**Comment by suricactus** [comment]
Hm, can we just stick to attachments here and use resources in the settings where required, just checked my code and I had 54 occurences of attachments already. Extra argument is that QFieldCloud already uses attachment in the files API and `attachmentDirs` setting, because I also believe the meaning is much more clear. 

If you strongly disagree, please add a commit that renames all the attachment names, so we stick to one name only.

**Review by suricactus — APPROVED** [review]
Looks great and very nice addition to the QField features list! I would wait before merging to see what would be the syntax for placeholders in the template.

File: layer.py
Comment (suricactus) [reply layer.py]: Let's wait on this before merge.

Comment (nirvn) [reply layer.py]: @suricactus , ya

File: layer.py
Comment (suricactus) [inline layer.py:97]: Actually a second though, can we make this a disctionary, so we have something like:

```
class AtachmentType(Enum):
	FILE = 0
        IMAGE = 1
       	WEB = 2
       	AUDIO = 3
       	VIDEO = 4

ATTACHMENT_EXPRESSIONS = {
        AttachmentType.FILE: "'files/{layername}_' || format_date(now(),'yyyyMMddhhmmsszzz') || '_#~filename~#'",
        AttachmentType.IMAGE: "'DCIM/{layername}_' || format_date(now(),'yyyyMMddhhmmsszzz') || '.#~extension~#'",
        AttachmentType.WEB: "",
        AttachmentType.AUDIO: "'audio/{layername}_' || format_date(now(),'yyyyMMddhhmmsszzz') || '.#~extension~#'",
        AttachmentType.VIDEO: "'video/{layername}_' || format_date(now(),'yyyyMMddhhmmsszzz') || '.#~extension~#'",
}
```

**Review by suricactus — CHANGES_REQUESTED** [review]

File: layer.py
Comment (suricactus) [reply layer.py:97]: Greatly improves readability and reduces these magic indeces floating around.

File: layer.py
Comment (suricactus) [inline layer.py:183]: We either get the `attachment_type` as an int already.

File: layer.py
Comment (suricactus) [inline layer.py:186]: Or we convert to `attachment_type` to `int` when accessing.

File: layer.py
Comment (suricactus) [reply layer.py:183]: See my suggestion in `https://github.com/opengisch/libqfieldsync/pull/41#discussion_r1089725178`, then it would be of type `AtachmentType`.

File: layer.py
Comment (suricactus) [inline layer.py:195]:
```suggestion
    def attachment_naming(self, field_name, attachment_type: AttachmentType) -> str:
```

I would do the conversion already in the caller.

File: layer.py
Comment (suricactus) [inline layer.py:198]:
```suggestion
            self.ATTACHMENT_EXPRESSIONS[attachment_type].format(
```

So this can be super straightforward.

File: layer.py
Comment (suricactus) [reply layer.py:195]: And since this will be annoying to do in the parent, I would simply add a utility function somewhere `get_attachment_type_by_int_value(value: int) -> AttachmentType`.

**Comment by nirvn** [comment]
@suricactus , done.

**Review by suricactus — APPROVED** [review]

**Pull request merged**
